lib/ansible/modules/network/netconf/netconf_config.py

```diff
@@ -270,6 +284,9 @@ def main():
             retkwargs=retkwargs,
             datastore=datastore,
         )
+        if changed and module.params['save']:
```

Comment:
This limitation is acceptable. The target of the module is NOT to do a stand-alone "copy running-config to startup-config". The aim is, that changes (if any) are persistently stored. At least the code proposal is doing this job.
The issue with your proposal would be, that the NETCONF server must support <get-config> on the <startup> datastore - which is not commonly supported by all netconf servers in the market. Also it is unclear, how the module returns the "changed" attribute - if edit-config results in unchanged config - while copy running to startup was still triggered, because of any differences.